### PR TITLE
vim update.py: allow different in and out files

### PR DESCRIPTION
**Pull request by teto**
#### Motivation for this change
I've got some extra vim plugins in my nixpkgs fork (that I haven't upstreamed for different reasons, not fixed, personal etc). When rebasing it often generates conflicts, so I end up using upstream generated.nix and at the end of the rebase call ./pkgs/misc/vim-plugins/update.py .
It can be boring to do so I would like to move this kind of personal list to an overlay.
I modified ./pkgs/misc/vim-plugins/update.py to accept optional input list and output file.
For instance
`./pkgs/misc/vim-plugins/update.py -i ~/dotfiles/config/nixpkgs/overlays/vim-plugins-names  -o toto.nix`

One issue is that I am not sure how to import these in my overlay. It would be nice if generated.nix could be implemented as an overlay so that the same update.py can generate code for nixpkgs and user overlays.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by Mic92** [comment]
Please re-format the file with black as well:

```
$ nix run nixpkgs.python3Packages.black -c black update.py
```

File: pkgs/misc/vim-plugins/update.py [hunk 1]
```diff
@@ -330,16 +331,32 @@ def generate_nix(plugins: List[Tuple[str, str, Plugin]]):
...
+    parser.add_argument(
+        "--input-names", "-i", dest="input_file", default=DEFAULT_IN,
+        help="A list of plugins in the form owner/repo"
+    )
+    parser.add_argument(
+        "--out", "-o", dest="outfile", default=DEFAULT_OUT,
+        help="Filename to save generated nix code"
+    )
+
+    args = parser.parse_args()
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/update.py]: Can you put the argument parsing in a function `parse_args()` that return `args` int the end?

Comment (teto) [reply pkgs/misc/vim-plugins/update.py]: done

**Comment by teto** [comment]
ran black on it and tested with the same command, worked fine.

**Comment by jonringer** [comment]
do you mind squashing and changing the commit message to:

```
vim update.py: allow different in and out files
```

**Comment by teto** [comment]
thanks for looking. done

**Pull request merged**